index.js

```diff
@@ -18,11 +24,19 @@ module.exports=function(invalidateTimeInMilliseconds,parameters){
         if(parameters && parameters.type){
             response.type(parameters.type);
         }
-        if (request.method == 'GET') {
+        if (request.method === 'GET') {
             cache.get(request.originalUrl,function(err,value){
                 if(value){
                     console.log('[CACHE] HIT: GET '+request.originalUrl);
-                    response.header('Cache-Control', 'private, no-cache, no-store, must-revalidate')
+
+                    var contentHash = getHash(value);
+                    if (request.get('ETag') === contentHash) {
+                        response.status(304).end();
+                        return true;
+                    }
+
+                    response.header('Cache-Control', 'private, no-cache');
```

Comment:
There are a few spots in poseidon where we set private, no-cache, no-store, must-revalidate (doing a search for no-cache in dh2o-poseidon/src shows 4 files and a no_cache middleware that is applied to the entire app.  Do any of these need to be changed?  

I guess this line will correct that header if poseidon has set it to something else so maybe it doesn't matter.